src/snmalloc/override/libc.h

```diff
@@ -0,0 +1,181 @@
+#pragma once
+
+#include "../global/global.h"
+
+#include <errno.h>
+#include <string.h>
+
+namespace snmalloc::libc
+{
+  /**
+   * Clang was helpfully inlining the constant return value, and
+   * thus converting from a tail call to an ordinary call.
+   */
+  static inline void* snmalloc_not_allocated = nullptr;
```

Comment:
Does marking this `const` re-enable the problematic inlining?

Reply:
I had a look at the code a bit more, and it no longer seems like it is a problem.

Reply:
Also ended up adding some SNMALLOC_FAST_PATH to force some inlining as `calloc` and `realloc` had introduced an additional tail call than needed.